core/jvm/src/main/scala/scalaz/zio/blocking/Blocking.scala

```diff
@@ -133,7 +133,11 @@ object Blocking extends Serializable {
                               Thread.interrupted // Clear interrupt status
                               Left(Cause.interrupt)
                             case t: Throwable =>
-                              Left(Cause.fail(t))
+                              try {
+                                Left(Cause.fail(t.asInstanceOf[E]))
```

Comment:
You need to add an implicit `ClassTag` for E & pattern match on `E` for this to ever fail. `t.asInstanceOf[E]` will never fail as `E` is erased to `AnyRef`

Reply:
Thank you 🙏